File: Nexus_Bilding_Site/src/types/fiscal/FiscalDocumentItem.ts
import { ITBISRate } from '../shared/ITBISRate';

export interface FiscalDocumentItem {
  productId?: string; // opcional para E41
  description: string;

  quantity: number;
  unitPrice: number;

  itbisRate: ITBISRate;
  itbisAmount: number;

  exemptionReason?: string;

  discount?: number;
  total: number;
}

File: Nexus_Bilding_API/nexus_bilding_API/nexus_bilding_api.core.domain/Entities/FiscalDocumentItem.cs
using nexus_bilding_api.core.domain.Common;
using nexus_bilding_api.core.domain.Enums;

namespace nexus_bilding_api.core.domain.Entities;

public class FiscalDocumentItem : BaseEntity
{
    public Guid FiscalDocumentId { get; set; }
    public FiscalDocument FiscalDocument { get; set; } = null!;

    public Guid? ProductId { get; set; }
    public Product? Product { get; set; }
    
    public string Description { get; set; } = string.Empty;
    
    public double Quantity { get; set; }
    public decimal UnitPrice { get; set; }
    
    public ItbisRate ItbisRate { get; set; }
    public decimal ItbisAmount { get; set; }
    
    public string? ExemptionReason { get; set; }
    
    public decimal? Discount { get; set; }
    public decimal Total { get; set; }
}
